File: Left Right Odd Count.py
'''
An array of N integers is given as the input to the program. For each integer the program must print the integer if the count of odd integers on the left side is equal to the count of odd integers on the right side. Else the program must print "-1" for that integer.

Solution

'''

n=int( input() )
l=list( map( int, input().split() ) ) 
m=[i%2 for i in l] 
o=0 
e=m[1:].count(1) 
if o==e:
    print(l[0], end=" ")
else:
    print("-1", end=" ")
for i in range(1,n): 
    if m[i-1]==1:
        o+=1 
    if m[i]==1:
        e-=1
    if o==e:
        print(l[i], end=" ")
    else:
        print("-1", end=" ")

File: Left Right Odd Count.c
/*

An array of N integers is given as the input to the program. For each integer the program must print the integer if the count of odd integers on the left side is equal to the count of odd integers on the right side. Else the program must print "-1" for that integer.

Solution:

*/

#include<stdio.h>
#include<stdlib.h>
int main()
{
    int a;
    scanf("%d\n",&a);
    int arr[a], r=0, l=0;
    for(int i=0; i<a; i++)
    {
        scanf("%d ", &arr[i]);
        if(arr[i]%2==1) r++;
    }
    for(int i=0; i<a; i++)
    {
        if(arr[i]%2==0)
	      {
            if(l==r) printf("%d ",arr[i]);
            else printf("-1 ");
        }
        else
	      {
            if(l==r-1) printf("%d ",arr[i]);
            else printf("-1 ");
    	  l++;
        r--;
        }
    }
}
